### PR TITLE
Replace deprecated commands

### PR DESCRIPTION
**Pull request by Yaty**
### Description

Replace deprecated commands

#### Related issues

- connect to #455 

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by Yaty** [comment]
Ok I think I didn't forget anything, should be good for you to review :+1:

**Comment by dhmlau** [comment]
@slnode test please

**Comment by Yaty** [comment]
This test is from the juggler : https://github.com/strongloop/loopback-datasource-juggler/blob/3e5c805418416cd7b4a2a328386282d56867ba4e/test/persistence-hooks.suite.js#L1322

It is also failling locally but I was wondering if we could no modify the juggler to expect `isNewInstance` ? Otherwise I can just define `isNewInstance` to null in this PR and the problem is solved :) What do you think ?

**Comment by virkt25** [comment]
Hi @Yaty, first off thank you for this amazing PR!! Much appreciated. I've reviewed the changes and everything looks good to me except we need to fix the one failing test. 

We can't modify the test in juggler as the tests are run against multiple connectors and ensure all connectors have the same behaviour. From looking at the error, we are expecting a `isNewInstance: true` but we are getting a `isNewInstance: [null]`. Given the test I think this is the relevant code that needs to be updated: https://github.com/strongloop/loopback-connector-mongodb/blob/4df3c63d0c4a0cb8d68339be7b06020cc69c1ddb/lib/mongodb.js#L569

Can you see what the `result` looks like. I suspect that the new `updateOne` operation is returning something different than what we had before and we just need to compute the value of `isNewInstance` based on the updated `result` we get back.

**Comment by Yaty** [comment]
Oh yes you're right ! I wasn't thinking about other connectors :)
Fixed ! The upsert option was needed.

**Comment by virkt25** [comment]
@slnode test please

**Comment by virkt25** [comment]
@slnode test please

**Comment by virkt25** [comment]
@slnode test please

File: test/mongodb.test.js [hunk 1]
```diff
@@ -805,8 +805,8 @@ describe('mongodb connector', function() {
     ) {
       Post.find(function(err, results) {
         events.should.eql([
-          'before execute insert',
-          'after execute insert',
+          'before execute insertOne',
+          'after execute insertOne',
```
Comment (bajtos) [inline test/mongodb.test.js]: I think this is a breaking change from the point of loopback-connector-mongodb users.

An application installing `before execute` or `after execute` hook an relying on `ctx.req.command` value `insert` will stop working after upgrade.

IMO, this change should be released as semver-major.

/cc @virkt25  @dhmlau @raymondfeng

Comment (virkt25) [reply test/mongodb.test.js]: Had a discussion with @raymondfeng and we came to the agreement that the minor release for this is fine as `mongodb` never bumped a major version for this release. 

Should there be an issue with this impacting users in a negative way we can look into reverting this commit and releasing a new minor version -- followed by a new major version containing this fix. Haven't seen any related issues thus far. 😄

Comment (virkt25) [reply test/mongodb.test.js]: After reading up some more on connector hooks I changed my mind, discussed this further and have opened a PR to address this.

**Pull request merged**